# Professor Adams - Financial Risk Management (FRE 6123) 
# Homework #3

### README

---
---
### Part 1 Set up Environment

In [1]:
import math
import pandas as pd
import numpy as np
import math
from dataclasses import dataclass
from typing import List, Tuple

---
---

### Part 2 Answer

#### Question 1 

In [4]:
# Credit risk metrics for a 4-year bullet loan

principal = 50_000_000.0  # USD
ytm = 0.03                # annual compounding
T = 4                     # years
recovery = 0.60
annual_pod = 0.0175       # per-year probability of default (conditional hazard)

# (a) Maximum exposure (just before maturity), bullet structure
ead_max = principal * ((1 + ytm) ** T)

# (b) Maximum Loss Given Default
lgd_max = (1 - recovery) * ead_max

# (c) Anticipated default rate in Year 4 (marginal PD in the 4th year)
survival_to_year4 = (1 - annual_pod) ** 3
marginal_pd_year4 = annual_pod * survival_to_year4

print("Max Expected Exposure (EAD_max) =", format(ead_max, ".6f"))
print("Max LGD =", format(lgd_max, ".6f"))
print("Survival to start of Year 4 =", format(survival_to_year4, ".6f"))
print("Anticipated default rate in Year 4 =", format(marginal_pd_year4, ".6f"))


Max Expected Exposure (EAD_max) = 56275440.500000
Max LGD = 22510176.200000
Survival to start of Year 4 = 0.948413
Anticipated default rate in Year 4 = 0.016597


#### Results (rounded)

* **( \mathrm{EAD}_{\max} \approx $56{,}275{,}440.50 )**

* **( \mathrm{LGD}_{\max} \approx $22{,}510{,}176.20 )**

* **Survival to start of Year 4 ( \approx 0.9484134 )**

* **Marginal PD in Year 4 ( \approx 1.6597% )**

#### Interpretation 

**(a) Maximum expected exposure (bullet loan)**
$$
\mathrm{EAD}_{\max} = \text{Principal} \times (1+\text{YTM})^{T} = 50\,\text{m} \times (1.03)^4 \approx \text{\$56.275440}\,\text{m}
$$

**(b) Maximum loss given default**
$$
\mathrm{LGD}_{\max} = (1-\text{Recovery}) \times \mathrm{EAD}_{\max} = 0.40 \times 56.275440\,\text{m} \approx \text{\$22.510176}\,\text{m}
$$

**(c) Anticipated default rate in Year 4 (marginal PD)**
$$
\Pr[\text{default in Y4}] = \text{POD} \times (1-\text{POD})^3 = 0.0175 \times (1-0.0175)^3 \approx 0.016597\ (\text{i.e., }1.659700\%)
$$

---
---